contracts/contracts/strategies/balancer/BaseAuraStrategy.sol

```diff
@@ -0,0 +1,159 @@
+import { IERC20 } from "../../utils/InitializableAbstractStrategy.sol";
+import { IERC4626 } from "../../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+import { IRewardStaking } from "../IRewardStaking.sol";
+
+import "hardhat/console.sol";
+
+abstract contract BaseAuraStrategy is BaseBalancerStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    address internal auraRewardPoolAddress;
+    address internal auraRewardStakerAddress;
+    uint256 internal auraDepositorPTokenId;
+    // renamed from __reserved to not shadow BaseBalancerStrategy.__reserved,
+    int256[50] private __reserved_2;
+
+    struct InitConfig {
+        address platformAddress; // platformAddress Address of the Balancer's pool
+        address vaultAddress; // vaultAddress Address of the vault
+        address auraRewardPoolAddress; // auraRewardPoolAddress Address of the Aura rewards pool
+        address auraRewardStakerAddress; // auraRewardStakerAddress Address of the Aura rewards staker
+        uint256 auraDepositorPTokenId; // auraDepositorPTokenId Address of the Aura rewards staker
+        bytes32 balancerPoolId; // balancerPoolId bytes32 poolId
+    }
+
+    /**
+     * Initializer for setting up strategy internal state. This overrides the
+     * InitializableAbstractStrategy initializer as Balancer's strategies don't fit
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Address of BAL & AURA
+     * @param _assets Addresses of supported assets. MUST be passed in the same
+     *                order as returned by coins on the pool contract, i.e.
+     *                WETH, stETH
+     * @param _pTokens Platform Token corresponding addresses
+     * @param initConfig additional configuration
+     */
+    function initialize(
+        address[] calldata _rewardTokenAddresses, // BAL & AURA
+        address[] calldata _assets,
+        address[] calldata _pTokens,
+        InitConfig calldata initConfig
+    ) external onlyGovernor initializer {
+        auraRewardPoolAddress = initConfig.auraRewardPoolAddress;
+        auraRewardStakerAddress = initConfig.auraRewardStakerAddress;
+        auraDepositorPTokenId = initConfig.auraDepositorPTokenId;
+        pTokenAddress = _pTokens[0];
+        maxWithdrawalSlippage = 1e15;
+        maxDepositSlippage = 1e15;
+        balancerPoolId = initConfig.balancerPoolId;
+        IERC20[] memory poolAssets = getPoolAssets();
+        uint256 assetsLength = _assets.length;
+        require(
+            poolAssets.length == assetsLength,
+            "Pool assets and _assets should be the same length."
+        );
+        for (uint256 i = 0; i < assetsLength; ++i) {
+            (address asset, ) = fromPoolAsset(address(poolAssets[i]), 0);
+            require(
+                _assets[i] == asset,
+                "Pool assets and _assets should all have the same numerical order."
```

Comment:
This doesn't mater that much as it's in the `initialize` function that is only run once, but its better to keep error strings to 31 or less characters so they fit in a 256-bit word. The extra byte stores the string length.

Reply:
got it 👍 useful info